packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -134,13 +134,59 @@ export type ScopedSchemaName<
+ *
+ * Note:
+ * POJO stands for Plain Old JavaScript Object.
+ * This means an object that works like a `{}` style object literal.
+ * In this case it means the prototype is `Object.prototype` and acts like a set of key value pairs (data, not methods).
+ * The usage below generalizes this to include array and map like objects as well.
+ *
+ * There are two ways to use these APIs:
+ * |                     | Customizable | POJO Emulation |
+ * | ------------------- | ------------ |--------------- |
+ * | Declaration         | `class X extends schemaFactory.object("x", {}) {}` | `const X = schemaFactory.object("x", {}); type X = NodeFromSchema<typeof X>; `
+ * | Allows adding "local" (non-persisted) members | Yes. Members (including methods) can be added to class.        | No. Attempting to set non-field members will error. |
+ * | Prototype | The user defined class | `Object.prototype`, `Map.prototype` or `Array.prototype` depending on node kind |
+ * | Structurally named Schema | Not Supported | Supported |
+ * | Explicitly named Objects | Supported | Supported |
+ * | Explicitly named Maps and Arrays | Supported: Both declaration approaches can be used | Not Supported |
+ * | node.js assert.deepEqual | Compares like class instances: equal to other nodes of the same type with the same content, including custom local fields. | Compares like plain objects: equal to plain JavaScript objects with the same fields, and other nodes with the same fields, even if the types are different. |
+ * | Intellisense | Shows and links to user defined class by name: `X` | Shows internal type generation logic: `object & TreeNode & ObjectFromSchemaRecord<{}> & WithType<"test.x">` |
+ * | Recursion | Supported with special declaration patterns. | Unsupported: Generated d.ts files replace recursive references with `any`, breaking use of recursive schema across compilation boundaries |
+ *
+ * Note that while "POJO Emulation" nodes act a lot like POJO objects, they are not true POJO objects:
+ *
+ * - Adding new arbitrary fields will error, as well some cases of invalid edits.
+ * - They are implemented using proxies.
+ * - They have state that is not exposed via enumerable own properties, including a {@link TreeNodeSchema}.
+ * This makes libraries like node.js `assert.deepEqual` fail to detect differences in type.
+ * - Assigning members has side effects (in this case editing the persisted/shared tree).
+ * - Not all operations implied by the prototype will work correctly: stick to the APIs explicitly declared in the TypeScript types.
+ *
+ * @privateRemarks
+ * Its perfectly possible to make `POJO Emulation` mode (or even just hiding the prototype) selectable even when using the custom user class declaration syntax.
+ * When doing this, its still possible to make `instanceof` perform correctly.
+ * Allowing of (or banning) custom/out of schema properties on the class is also possible in both modes: it could be orthogonal.
```

Comment:
Nit
```suggestion
 * Allowing (or banning) custom/out-of-schema properties on the class is also possible in both modes: it could be orthogonal.
```